### PR TITLE
New colors set and css fixes

### PR DESCRIPTION
**Pull request by skjnldsv**
This pr make the calendar app use the core colour generator to get an unified palette.

@jancborchardt 
ref #344
capture d ecran_2016-05-05_12-05-41

### CODE REVIEW

File: js/app/directives/colorpickerDirective.js [hunk 1]
```diff
@@ -21,12 +21,65 @@
  *
  */
 
+// expected hue range: [0, 360)
+// expected saturation range: [0, 1]
+// expected lightness range: [0, 1]
+var hslToRgb = function(hue, saturation, lightness){
+  // based on algorithm from http://en.wikipedia.org/wiki/HSL_and_HSV#Converting_to_RGB
+  if( hue == undefined ){
```
Comment (georgehrke) [inline js/app/directives/colorpickerDirective.js]: please use tabs instead of spaces :)

Comment (skjnldsv) [inline js/app/directives/colorpickerDirective.js]: Damn copy/paste :p

Comment (skjnldsv) [inline js/app/directives/colorpickerDirective.js]: Wait a minute, the entire project uses space and not tabs, should I change the whole file @georgehrke ?

Comment (georgehrke) [inline js/app/directives/colorpickerDirective.js]: yes, please change the entire file

**Comment by jancborchardt** [comment]
I'm a bitnon the femce about the colors – the green looks too garish and the orange too flat. … but yeah they should come from the core generator.

Any way to shift that orange and green a bit to make it a tad nicer?

cc @georgehrke @raghunayyar @owncloud/designers check it out.

**Comment by skjnldsv** [comment]
We could also just add a colorwheel and remove the 9 colours ? :D

**Comment by raghunayyar** [comment]
@jancborchardt @skjnldsv just a suggestion, we can use HSLA colours and keep the saturation intact and change hue. The colors generated will probably will be on the same palette? What do you guys think? I think it works otherwise just the colours look too dull.

**Comment by georgehrke** [comment]
@raghunayyar We need to store RGB in the CalDAV property

**Comment by skjnldsv** [comment]
@raghunayyar we're already doing that in core.
We're using HSL and only changing the hue.

**Comment by raghunayyar** [comment]
> @raghunayyar We need to store RGB in the CalDAV property

Can't we convert the HSLA to RGB in that case, when we are storing the colorcode?

**Comment by skjnldsv** [comment]
@raghunayyar have you read this pull request??

**Comment by raghunayyar** [comment]
@skjnldsv Lol, I am so sorry. Just reviewed the PR. I got quite emotional earlier seeing the colors being too light. Otherwise :+1:

**Comment by skjnldsv** [comment]
Ahaha, no problem! :p
I was wondering if you didn't had the wrong pr! ^^

The colors being too light comes from the core generator then. But now that you said that, I maybe had forgotten to update the s&l values based on the luminosity. I need to take a look!

**Comment by skjnldsv** [comment]
Here we go. Rebased and ready to be merged.
I'm now using hardcoded values to generate the full hsl values, which I previously wasn't doing and therefore was ignoring the luminosity correction from core.

Now we're all set and ready to go!
capture d ecran_2016-05-10_17-48-27

If we want to shift the colors, we can also do that :)

**Comment by skjnldsv** [comment]
Shifting:
00: calendar owncloud0
10: calendar owncloud1
20: calendar owncloud2
30: calendar owncloud3

@jancborchardt

**Comment by raghunayyar** [comment]
:+1:

**Comment by skjnldsv** [comment]
Which one do you prefer? :)

**Comment by georgehrke** [comment]
I'd go with 00

**Comment by aspdye** [comment]
I am missing a "real" blue and some other natural, but i think this is nothing we can change here :cry: 
I'd also vote for 00.

**Comment by MorrisJobke** [comment]
> I am missing a "real" blue and some other natural, but i think this is nothing we can change here

Why not. This sounds reasonable. :) @skjnldsv is this somehow possible to accomplish?

**Comment by skjnldsv** [comment]
You know what, here's what I'm proposing :)
We've got a better blue, and some good and various colours!
capture d ecran_2016-05-11_10-14-25

@aspdye

**Comment by MorrisJobke** [comment]
> We've got a better blue, and some good and various colours!

I really like it :)

**Comment by skjnldsv** [comment]
And here's a gift:
sans titre

**Comment by raghunayyar** [comment]
We should get this in asap! :D

**Comment by georgehrke** [comment]
- [ ] This pr makes the spinner in front of the calendar name huge

**Comment by georgehrke** [comment]
Using chromium 52:

calendar - owncloud chromium today at 12 30 49 pm

The colors are smaller (why?)
For some reason it shows a different icon for random, also clicking it does nothing

**Comment by aspdye** [comment]
I like the colors of chromium 52 :+1:

**Comment by skjnldsv** [comment]
Hum, I may have pushed the wrong folder! :p 
I'm going to fix this when I get home ^^

> The colors are smaller (why?)

Because there is a 10th element added to the list, so it has to take the full width.

> This pr makes the spinner in front of the calendar name huge

See: #509, this is not related to this pr. I didn't changed anything related to the loader here.

EDIT: @georgehrke, I cheated, I did it from work! Could you check? :3

**Comment by skjnldsv** [comment]
Hum, should I edit `js/public/app.js` too?
Or is this file generated like the contacts app? I find it strange to have the same functions on two separate location :)

**Comment by georgehrke** [comment]
You have to build the app.js with `grunt build` inside js/

Please excuse my brevity and typos.

**Comment by skjnldsv** [comment]
@georgehrke Peeeerfect! That's why this wasn't working for you :)
I was assuming the file would be compiled like owncloud/contacts. 

Thank you!

**Comment by georgehrke** [comment]
How did you develop it without changing the app.js? ^^

Please excuse my brevity and typos.

**Comment by skjnldsv** [comment]
@georgehrke I compiled in my head! :D 
Just kidding, I manually edited the app.js because the makefile isn't compatible with my dev env.

**Comment by skjnldsv** [comment]
@raghunayyar, @jancborchardt, @tomneedham, @georgehrke, @MorrisJobke Review please?

**Comment by georgehrke** [comment]
👍

**Comment by georgehrke** [comment]
really like it! :)

**Comment by skjnldsv** [comment]
Thanks! :)

**Comment by enoch85** [comment]
Nice work @skjnldsv :+1:

**Comment by georgehrke** [comment]
rgbToHex is pretty broken 🙈
`rgbToHex(0,100,0)` gives me `#0640`

will fix it in https://github.com/owncloud/calendar/pull/543

**Comment by skjnldsv** [comment]
@georgehrke is it due to a missing str_pad?
Don't recall the js function, but it may be due to the 0 converted to integers! :)

**Comment by skjnldsv** [comment]
Yep, here we go: 
```
function componentToHex(c) {
    var hex = c.toString(16);
    return hex.length == 1 ? "0" + hex : hex;
}

/*
 * Convert rgb array to hex string
 */
var rgbToHex = function(r, g, b) {
	'use strict';
	if(Array.isArray(r)) {
		[r, g, b] = r;
	}
	return '#' + componentToHex(parseInt(r, 10)) + componentToHex(parseInt(g, 10)) + componentToHex(parseInt(b, 10));
};
```

**Comment by jancborchardt** [comment]
Btw again @skjnldsv really good work! :)

**Comment by skjnldsv** [comment]
Thank you jan! :)

**Pull request merged**